File: library/src/main/java/me/proxer/library/entity/user/TopTenEntry.java
package me.proxer.library.entity.user;

import com.squareup.moshi.Json;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Value;
import me.proxer.library.entity.ProxerIdItem;
import me.proxer.library.enums.Category;
import me.proxer.library.enums.Medium;

import javax.annotation.Nullable;

/**
 * Entity representing a single Topten entry.
 *
 * @author Ruben Gees
 */
@Value
@EqualsAndHashCode(onParam = @__({@Nullable}))
public class TopTenEntry implements ProxerIdItem {

    /**
     * Returns the id.
     */
    @Getter(onMethod = @__({@Override}))
    @Json(name = "eid")
    private String id;

    /**
     * Returns the name.
     */
    @Json(name = "name")
    private String name;

    /**
     * Returns the category.
     */
    @Json(name = "kat")
    private Category category;

    /**
     * Returns the medium.
     */
    @Json(name = "medium")
    private Medium medium;
}

File: library/src/main/kotlin/me/proxer/library/entity/user/TopTenEntry.kt
package me.proxer.library.entity.user

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import me.proxer.library.entity.ProxerIdItem
import me.proxer.library.enums.Category
import me.proxer.library.enums.Medium

/**
 * Entity representing a single Topten entry.
 *
 * @property name The name.
 * @property category The category.
 * @property medium The medium.
 *
 * @author Ruben Gees
 */
@JsonClass(generateAdapter = true)
data class TopTenEntry(
    @Json(name = "eid") override val id: String,
    @Json(name = "name") val name: String,
    @Json(name = "kat") val category: Category,
    @Json(name = "medium") val medium: Medium
) : ProxerIdItem
